### PR TITLE
Trash notes screen

### PR DESCRIPTION
**Pull request by gonzalovelasco**
Screenshot 2023-08-08 at 19 33 25

Screenshot 2023-08-09 at 15 13 50

Screenshot 2023-08-08 at 19 33 56

### CODE REVIEW

File: lib/screens/main/components/side_rail.dart [hunk 1]
```diff
@@ -8,11 +8,9 @@ void onSetting(BuildContext context) {
 class SideRail extends StatelessWidget {
   const SideRail({
     Key? key,
-    required this.addNote,
     required this.onMenu,
   }) : super(key: key);
 
-  final VoidCallback addNote;
```
Comment (matthewwong525) [inline lib/screens/main/components/side_rail.dart:15]: great job cleaning this up!

**Review by matthewwong525 — APPROVED** [review]

**Review by matthewwong525 — CHANGES_REQUESTED** [review]
This is a pretty big feature, could you add some tests? for example:
- go to trash > see empty > go to main delete note > go to trash > see note
- (and another for restore note)

just cover these 2 main use cases in tests. thanks and great work. apart from tests things look good

**Comment by gonzalovelasco** [comment]
> This is a pretty big feature, could you add some tests? for example:
> 
> * go to trash > see empty > go to main delete note > go to trash > see note
> * (and another for restore note)
> 
> just cover these 2 main use cases in tests. thanks and great work. apart from tests things look good

Tests added a45c6070a7ac7e7dd401c1f4778c5749cbc2fe76

**Review by matthewwong525 — APPROVED** [review]
LGTM

**Pull request merged**